src/poetry/utils/password_manager.py

```diff
@@ -180,11 +180,8 @@ def get_pypi_token(self, repo_name: str) -> str | None:
         :param repo_name:  Name of repository.
         :return: Returns a token as a string if found, otherwise None.
         """
-        return self._config.get(f"pypi-token.{repo_name}", self.keyring.get_password(repo_name, "__token__"))
-        return (
-            token
-            if token is not None
-            else self.keyring.get_password(repo_name, "__token__")
+        return self._config.get(
+            f"pypi-token.{repo_name}", self.keyring.get_password(repo_name, "__token__")
```

Comment:
Time to spoil the fun: I actually liked having keyring being called lazily.

Reply:
could you explain or just suggest the change because I don't really know what you mean @Darsstar

Reply:
```
token = self._config.get(f"pypi-token.{repo_name}")
if token:
    return token

return self.keyring.get_password(repo_name, "__token__")
```
That way we only query keyring if the config does not have a token for us.
If Python had a get variant which accepted a Callable that returns a default value... But alas, lamda's are not considered Pythonic.

Reply:
@Darsstar please check now